phpBB/phpbb/avatar/driver/upload.php

```diff
@@ -161,6 +161,16 @@ public function process_form($request, $template, $user, $row, &$error)
 			return false;
 		}
 
+		$ext = substr(strrchr($row['avatar'], '.'), 1);
+		if ($ext && $ext !== $file->get('extension'))
+		{
+			$filename = $this->phpbb_root_path . $this->config['avatar_path'] . '/' . $this->config['avatar_salt'] . '_' . $row['id'] . '.' . $ext;
```

Comment:
I know this is the same code, but unfortunately, it would conflict with another ticket I have submitted: 
https://github.com/phpbb/phpbb/pull/3738

If I called the delete method from here, there would be no way to differentiate a "genuine" call to delete (that would require an event being triggered) with a code reuse use (this call) that would require the event is not called, as it would have been already called in this method (around L153 here).

It could be done by passing an extra parameter to the delete function that would flag this use, or by reconsidering the way to work the events.

But please, see both tickets in conjunction, and provide your opinion/feedback.  Will be happy to implement whatever it is that we decide here.